source/distributed-omit.d.ts

```diff
@@ -0,0 +1,80 @@
+import type {KeysOfUnion} from './keys-of-union';
+
+/**
+Omits keys from a type, distributing the operation over a union.
+
+TypeScript's `Omit` doesn't distribute over unions, which causes the following situation:
```

Comment:
It would be useful to have a description in text, not just examples. Examples take longer to understand.

Here's a suggested text. Can you check that it's correct?
```suggestion
TypeScript's `Omit` doesn't distribute over unions, leading to the erasure of unique properties from union members when omitting keys. This creates a type that only retains properties common to all union members, making it impossible to access member-specific properties after the Omit. Essentially, using `Omit` on a union type merges the types into a less specific one, hindering type narrowing and property access based on discriminants. This type solves that.
```

Feel free to edit it as you see fit.

Reply:
Agreed